EnableJupyterDisplay.swift

```diff
@@ -0,0 +1,117 @@
+import Cryptor
+import Foundation
+
+struct Header: Codable {
+    let msg_id: String
+    let username: String
+    let session: String
+    var date: String {
+        let currentDate = Date()
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSZZZZZ"
+        formatter.timeZone = TimeZone(secondsFromGMT: 0)
+        formatter.locale = Locale(identifier: "en_US_POSIX")
+        return formatter.string(from: currentDate)
+    }
+    let msg_type: String
+    let version: String
+
+    public init(msg_id: String = UUID().uuidString, username: String = "kernel", session: String, msg_type: String = "display_data", version: String = "5.2") {
+        self.msg_id = msg_id
+        self.username = username
+        self.session = session
+        self.msg_type = msg_type
+        self.version = version
+    }
+
+    public func toJSON() -> String {
+        do {
+            let jsonData = try JSONEncoder().encode(self)
+            let jsonString = String(data: jsonData, encoding: .utf8)!
+            return jsonString
+        }
+        catch {
+            return "{}"
+        }
+    }
+}
+
+struct Message {
+    var msgType: String = "display_data"
+    var delimiter = "<IDS|MSG>"
+    var key: String = ""
+    var header: Header
+    var metaData: String = "{}"
+    var content: String = "{}"
+    var hmacSignature: String {
+        let data = Data((header.toJSON()+parentHeader+metaData+content).utf8)
+        let keyData = Data(key.utf8)
+        let dataHex = data.map{String(format: "%02x", $0)}.joined()
```

Comment:
Always have spaces around the `{}` that make the closure.

`data.map { String(format: "%02x", $0) }.joined()`